zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java

```diff
@@ -202,46 +202,59 @@ public void testAngularDisplay() throws InterruptedException{
-        By.xpath(getParagraphXPath(1) + "//div[starts-with(@id, 'p') and contains(@id, 'text')]/div")));
-
-    // reset dependency
-    interpreterLink.click();
-    sparkEditBtn = pollingWait(By.xpath("//div[h3[text()[contains(.,'spark')]]]//button[contains(.,'edit')]"),
-        MAX_BROWSER_TIMEOUT_SEC);
-    sparkEditBtn.click();
-    WebElement testDepRemoveBtn = driver.findElement(By.xpath("//tr[descendant::text()[contains(.,'" +
-        artifact + "')]]/td[3]/div"));
-    sleep(5000, true);
-    testDepRemoveBtn.click();
-    driver.findElement(By.xpath("//button[contains(.,'Save')]")).submit();
-    driver.switchTo().alert().accept();
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+    try {
+      // navigate to interpreter page
+      WebElement interpreterLink = driver.findElement(By.linkText("Interpreter"));
+      interpreterLink.click();
+
+      // add new dependency to spark interpreter
+      WebElement sparkEditBtn = pollingWait(By.xpath("//div[h3[text()[contains(.,'spark')]]]//button[contains(.,'edit')]"),
+          MAX_BROWSER_TIMEOUT_SEC);
+      sparkEditBtn.click();
+      WebElement depArtifact = driver.findElement(By.xpath("//input[@ng-model='setting.depArtifact']"));
+      String artifact = "org.apache.commons:commons-csv:1.1";
+      depArtifact.sendKeys(artifact);
+      driver.findElement(By.xpath("//button[contains(.,'Save')]")).submit();
+      driver.switchTo().alert().accept();
+
+      driver.navigate().back();
+      createNewNote();
+
+      // wait for first paragraph's " READY " status text
+      waitForParagraph(1, "READY");
+
+      WebElement paragraph1Editor = driver.findElement(By.xpath(getParagraphXPath(1) + "//textarea"));
+
+      paragraph1Editor.sendKeys("import org.apache.commons.csv.CSVFormat");
+      paragraph1Editor.sendKeys(Keys.chord(Keys.SHIFT, Keys.ENTER));
+      waitForParagraph(1, "FINISHED");
+
+      // check expected text
+      assertTrue(waitForText("import org.apache.commons.csv.CSVFormat",
+          By.xpath(getParagraphXPath(1) + "//div[starts-with(@id, 'p') and contains(@id, 'text')]/div")));
+
+      //delete created notebook for cleanup.
+      deleteTestNotebook(driver);
+      sleep(1000, true);
+
+      // reset dependency
+      interpreterLink.click();
+      sparkEditBtn = pollingWait(By.xpath("//div[h3[text()[contains(.,'spark')]]]//button[contains(.,'edit')]"),
+          MAX_BROWSER_TIMEOUT_SEC);
+      sparkEditBtn.click();
+      WebElement testDepRemoveBtn = driver.findElement(By.xpath("//tr[descendant::text()[contains(.,'" +
+          artifact + "')]]/td[3]/div"));
+      sleep(5000, true);
+      testDepRemoveBtn.click();
+      driver.findElement(By.xpath("//button[contains(.,'Save')]")).submit();
+      driver.switchTo().alert().accept();
+    } catch (Exception e) {
+      LOG.error("Exception in ZeppelinIT while testSparkInterpreterDependencyLoading ", e);
+      File scrFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
+
```

Comment:
You might want to just `throw e` - exception stack should already have the test name?